<a href="https://colab.research.google.com/github/mr-zero-000/Statistical-Learning-e23034/blob/main/Assignment_1%20/%20Assignment_1_Q1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Q1) The Probability of Shooting at a Targe**t

A point is chosen at random from a disk of radius $R=10$. Let $A$ be the event that the point lies within $1$ unit of the boundary. Therefore the distance $r$ from the center satisfies $R-1 \le r \le R$, or $9 \le r \le 10$.

### Model 1: Point is uniformly distributed with respect to area.

In [1]:
import math

R = 10

# Total area of the disk
area_total = math.pi * R**2

# Area of the region where the point lies within 1 unit of the boundary (an annulus)
# This is the area of the disk of radius R minus the area of the disk of radius (R-1)
area_A_model1 = math.pi * R**2 - math.pi * (R - 1)**2

P_A_model1 = area_A_model1 / area_total

print(f"Total area of the disk: {area_total:.2f}")
print(f"Area of event A (within 1 unit of boundary): {area_A_model1:.2f}")
print(f"P(A) for Model 1 (uniform area): {P_A_model1:.4f}")

Total area of the disk: 314.16
Area of event A (within 1 unit of boundary): 59.69
P(A) for Model 1 (uniform area): 0.1900


### Model 2: Distance to the point from the center is chosen uniformly from $[0,10]$, and the direction is chosen independently and uniformly from $[0,2\pi)$.

In [2]:
# In this model, the radial distance r is uniformly distributed in [0, 10].
# The event A is that the point lies within 1 unit of the boundary, i.e., 9 <= r <= 10.

# The probability of r falling into an interval [a, b] is (b - a) / (max_r - min_r)
# Here, min_r = 0, max_r = 10, a = 9, b = 10.

P_A_model2 = (10 - 9) / (10 - 0)

print(f"P(A) for Model 2 (uniform radial distance): {P_A_model2:.4f}")

P(A) for Model 2 (uniform radial distance): 0.1000


The probabilities differ because the point is selected using two different distributions:

**1. Uniform with respect to area:**
Every part of the disk’s area is equally likely. Since outer rings have larger area than inner rings, points are more likely to lie near the boundary.
For the outer ring (radius 9 to 10):
$$
P(A)=\frac{\pi(10^2-9^2)}{\pi(10^2)}=\frac{19}{100}=0.19
$$

**2. Uniform with respect to radius:**
Each radial distance from 0 to 10 is equally likely. So the probability depends only on radial length, not area.
For radius interval 9 to 10:
$$
P(A)=\frac{10-9}{10}=0.10
$$

**Conclusion:**
The uniform area model gives a higher probability because outer regions cover more area, while the uniform radius model treats all radial intervals equally.